### ISSUE TITLE
[FEATURE] Allow modular model saving/loading

### ISSUE THREAD

**Issue by Larwive** [Larwive/Enhanced-World-Model#31]
## Summary
With modular model saving/loading, we will be able to load specific parts of pretrained models separately.

## Problem Statement
It makes the project even more modular and convenient to use.

## Proposed Solution
Add options to choose which part of a saved model to load, without forgetting to check compatibility.

## Alternatives Considered
None.

## Implementation Plan

This feature can be broke down into 1 PR per training loop as they have different needs.

- [ ] Step 1: Ensure enough information is saved for modular loading, especially the environment name.
- [ ] Step 2: Breakdown model loading with choice by user.

The user will specify the parts to load with (in order of priority, a top loaded part can be overwritten by a lower one):
- `--load` for the entire model
- `--load-vision` for the vision part
- `--load-memory` for the memory part
- `--load-controller` for the controller part
- `--load-reward-predictor` for the reward predictor

## Additional Context
See https://docs.pytorch.org/tutorials/beginner/saving_loading_models.html and existing code about saving/loading.

## Acceptance Criteria
Safe (no code injection possible) saving/loading of models.
